chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -531,6 +532,19 @@ async def get_puzzle_info(self, nft_id: bytes32) -> PuzzleInfo:
         else:
             return puzzle_info
 
+    async def sign_message(self, message: bytes, nft: NFTCoinInfo) -> Tuple[G1Element, G2Element]:
+        uncurried_nft = UncurriedNFT.uncurry(*nft.full_puzzle.uncurry())
+        if uncurried_nft is not None:
+            p2_puzzle = uncurried_nft.p2_puzzle
+            puzzle_hash = p2_puzzle.get_tree_hash()
+            pubkey, private = await self.wallet_state_manager.get_keys(puzzle_hash)
+            synthetic_secret_key = calculate_synthetic_secret_key(private, DEFAULT_HIDDEN_PUZZLE_HASH)
+            synthetic_pk = synthetic_secret_key.get_g1()
+            prefix = f"\x18Chia Signed Message:\n{len(message)}"
+            return synthetic_pk, AugSchemeMPL.sign(synthetic_secret_key, std_hash(prefix.encode("utf-8") + message))
```

Comment:
Is there a standard that we're basing this off of? Because it seems odd to me that the length of the message is being encoded in the prefix string as a sequence of characters. That is, if my message is "hello world!" is the expectation that the length (12) is represented as `'1'` and `'2'`?

I would normally expect a construction like the following, but I could be missing something...

`std_hash(len("Chia Signed Message:\n") + "Chia Signed Message:\n".encode("utf-8") + len(message))`

And for length values that don't fit in a single byte, I would assume (again, maybe incorrectly) network byte ordering.

Reply:
Pending on CHIP0002